src/pymatgen/io/vasp/inputs.py

```diff
@@ -744,7 +746,7 @@ def __setitem__(self, key: str, val: Any) -> None:
         """
         super().__setitem__(
             key.strip(),
-            type(self).proc_val(key.strip(), val.strip()) if isinstance(val, str) else val,
```

Comment:
I believe this is a subtle bug, where only `str` type would go through the `proc_val` preprocessor.

However there's another point that is unclear to me now, as the `proc_val` should be able to handle `list` keys as well (the `from_str` method): https://github.com/materialsproject/pymatgen/blob/32d9bebbebf24043bdf56683333053dcb1980d23/src/pymatgen/io/vasp/inputs.py#L867-L885

https://github.com/materialsproject/pymatgen/blob/32d9bebbebf24043bdf56683333053dcb1980d23/src/pymatgen/io/vasp/inputs.py#L847-L865